stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channels/ChannelListViewModel.kt

```diff
@@ -146,22 +176,19 @@ public class ChannelListViewModel(
      * Makes the initial query to request channels and starts observing state changes.
      */
     private suspend fun init() {
-        chatClient.globalState.user.map { it?.id }.filterNotNull().flatMapLatest {
-            searchQuery.combine(queryConfig) { query, config -> query to config }
-        }.collectLatest { (query, config) ->
-            val filter = createQueryChannelsFilter(config.filters, query)
-
-            val queryChannelsRequest = QueryChannelsRequest(
-                filter = filter,
-                querySort = config.querySort,
-                limit = channelLimit,
-                messageLimit = messageLimit,
-                memberLimit = memberLimit,
-            )
+        searchQuery.combine(queryConfigFlow) { query, config -> query to config }
+            .collectLatest { (query, config) ->
+                val queryChannelsRequest = QueryChannelsRequest(
```

Comment:
Sometime we get 2 emits here:
- The first one with NeutralFilterObject
- The second one with the correct default filter
Can we don something about that?

Reply:
Good point - we could filterNotNull on the `filter` flow and then just ignore any "requests" where the filter doesn't exist

Reply:
@bychkovdmitry added a filter! Can you check now? :]

Reply:
The issue is solved